### ISSUE TITLE
I2S Problem

### ISSUE THREAD

**Issue by wa8lbz** [Edzelf/ESP32Radio-V2#36]

**Comment by wa8lbz** [comment]
Since I am having on and off results with the VS1053, I am going to try I2S again. I am using a PCM5102A board connected as follows: BCK to GPIO 5, DIN to GPIO 32 and LCK to GPIO 4. I have changed the Config.h accordingly. See below. When I issued the PIO run -t upload command it fails. See below.
![I2S](https://user-images.githubusercontent.com/45833682/165823913-a465aba8-c1eb-4842-88c8-c7c158df1239.PNG)
![I2S-2](https://user-images.githubusercontent.com/45833682/165823945-8c7cb5d7-54ea-4d4d-b210-e0976b6653af.PNG)

**Comment by Edzelf** [comment]
Pio update?

**Comment by wa8lbz** [comment]
I updated PlatformIO. Same results.

**Comment by Edzelf** [comment]
When compiling, do you see this platform and these packages?  Check the version numbers.

```
PLATFORM: Espressif 32 (3.3.2) > DOIT ESP32 DEVKIT V1
..
PACKAGES:
 - framework-arduinoespressif32 3.10006.210326 (1.0.6)
 - tool-esptoolpy 1.30100.210531 (3.1.0)
 - tool-mkspiffs 2.230.0 (2.30)
 - toolchain-xtensa32 2.50200.97 (5.2.0)
```

**Comment by wa8lbz** [comment]
This what I see:
[What I See.txt](https://github.com/Edzelf/ESP32Radio-V2/files/8591688/What.I.See.txt)

**Comment by Edzelf** [comment]
Version is okay.  Change I2S_COMM_FORMAT_STAND_I2S to **I2S_COMM_FORMAT_I2S**.  I will update it.

**Comment by wa8lbz** [comment]
This is what I see after making the change:
![After Change](https://user-images.githubusercontent.com/45833682/165993179-01fbaae5-a4d7-45d2-b723-c601feebd586.PNG)
:

**Comment by Edzelf** [comment]
Can you find out out what is refering to "internal_dac"?  I cannot see any reference.
Compare your compiler output to the attached listing.
[compile.txt](https://github.com/Edzelf/ESP32Radio-V2/files/8592793/compile.txt)
.

**Comment by wa8lbz** [comment]
I compared your output to my output but could not find any reference to "internal_dac". Below is my output:
[My Output.txt](https://github.com/Edzelf/ESP32Radio-V2/files/8594482/My.Output.txt)

**Comment by wa8lbz** [comment]
This is one place I found reference to "inter_dac"
![Internal Dac](https://user-images.githubusercontent.com/45833682/166063466-e0c13eb4-ea8f-4fc1-a772-aaaaf69451dc.PNG)
.

**Comment by Edzelf** [comment]
Mm, that's an old version.  I replaced helixfuncs.h with the current version.

**Comment by wa8lbz** [comment]
The current version of helixfuncs.h allowed a successful upload. The I2S radio now plays. At this time there are still two outstanding problems to be resolved.

Problem #1: No matter how many times I change the default preset = xx, click on Save and click on restart, the radio starts up playing preset 00. This still happens when I power the radio down then power it back up. The radio is still playing preset 00. All of the other preset functions work as they should.

Problem #2: According to the debug output: 'SSID = xxxxxxxxx' should appear on page 0 line 5 (t4) of the Nextion display and 'IP= 192.168.0.13' should appear on page 0 line 6 (t5) of the display. But they don't. It would be very helpful to have the IP address appear on the display.

Ed, thanks for all your help.

**Comment by Edzelf** [comment]
Problem 1 should be corrected now.

**Comment by wa8lbz** [comment]
Yes, problem 1 is now solved. Looking forward to the resolution of problem 2.

**Comment by wa8lbz** [comment]
I get the error messages shown below when uploading the latest code. It looks similar to the error messages when older Helixfuncs.h file was used.
[Maybe Old Helixfuncs dot h.txt](https://github.com/Edzelf/ESP32Radio-V2/files/8681529/Maybe.Old.Helixfuncs.dot.h.txt)

**Comment by Edzelf** [comment]
Use  the last version of the espressif32 framework (pio update).

**Comment by wa8lbz** [comment]
I down graded to Espressif 4.2.0. Now I only get one error message during upload. See below.
![Error Now](https://user-images.githubusercontent.com/45833682/168159277-37f44dac-91c3-49e9-9bc1-66a14acafbae.PNG)

**Comment by Edzelf** [comment]
Do a "clean" and/or remove .pio and .vscode directories.

**Comment by wa8lbz** [comment]
Performed a Clean successfully. Removed .pio and .vscode directories. The upload gave the error message below.
![New Error](https://user-images.githubusercontent.com/45833682/168339053-481168fb-17d3-42cd-a0a8-3c49d4dfa3e8.PNG)

**Comment by wa8lbz** [comment]
I forgot to mention that the .pio directory comes back immediately after deleting it. Maybe this is normal. I tried an upload again and this time it was successful. The the radio just continues to reboot over and over again.

**Comment by wa8lbz** [comment]
Successful uploads but endless radio reboots. Below is the Serial Monitor logging:
![Seral Monitor](https://user-images.githubusercontent.com/45833682/168343602-0dff7ffb-883e-48de-85a6-1c67572f020d.PNG)

**Comment by Edzelf** [comment]
The traceback is about a crash of platformio caused by a failing serial to USB driver.  Could be the powersupply.

**Comment by armecozzi** [comment]
Sorry for not getting back to you sooner, but my Internet provider changed and am having all kinds of problems. You are right, the constant rebooting was the result of a faulty power supply. Everything is now working as it should. Now all the lines on page 0 of the display are shown. Suggestion, add a 4 or  5 second delay after page 0 is complete. This gives time to view what is on page 0 before it changes to page 1. Ed, thank you for all of your hard work.

**Comment by Edzelf** [comment]
Add some extra time, se line 1640..1643:
```
  #ifdef NEXTION
    delay ( 2000 ) ;                                    // Show for some time
    dsp_println ( "\f" ) ;                              // Select new page if NEXTION 
  #endif
```

**Comment by wa8lbz** [comment]
Made the appropriate changes in the code. Timing now good.

Ed, I want to thank you for all of your hard work. Thanks to you, all of the functions/hardware in my ESP32 Radio V2 are now working as they should.

**Comment by wa8lbz** [comment]
I think I have found a problem. The picture below shows the GPIO pins for the I@S interface. Notice the description of the data output pin as i2s_din_pin.  
![Main dot cpp picture 1](https://user-images.githubusercontent.com/45833682/170338030-0ffcd4e7-798a-4e0a-b349-873e6ec2d000.PNG)

Now look at the picture below showing the Preferences like... here the data output pin is i2s_bin_pin.
![Manual](https://user-images.githubusercontent.com/45833682/170338607-5602f499-b9ee-485a-9d78-08e890fae06c.PNG)

One of the descriptions is wrong.
